### ISSUE TITLE
Add support for UDP

### ISSUE THREAD

**Issue by knholland** [logzio/logzio-nodejs#12]
Is it possible to add support for port 5050 to allow UDP?
http://support.logz.io/support/solutions/articles/6000112383-how-do-i-ship-json-logs-to-loz-io-via-udp-

**Comment by asafm** [comment]
Hi Katie,

That's possible. I'd be happy to know what's your use case here? UDP is
some serious limitations compared with the Bulk HTTP protocol we use.

**Comment by gillyb** [comment]
@khowen Is this still relevant to you ?

**Comment by knholland** [comment]
Yes, thank you for the follow up! We found this [link](http://support.logz.io/support/solutions/articles/6000112383-how-do-i-ship-json-logs-to-loz-io-via-udp-) that specifies we would need support for port 5050 for UDP. We are looking for this to support projects that currently use UDP and may not be able to move away from UDP in the near future.

**Comment by gillyb** [comment]
@khowen UDP support was added quite a while ago
(In this commit specifically : https://github.com/logzio/logzio-nodejs/commit/ef4ca8b73d38ea548cde7be5396d47394cc38391)

Are you using it ? Is everything working ok?

**Comment by sempasha** [comment]
There are some issues in UDP support.

```javascript
LogzioLogger.prototype._sendMessagesUDP = function() {
    var messagesLength = this.messages.length;

    var udpSentCallback = function(err, bytes) {
        if (err) {
            this._debug('Error while sending udp packets. err = ' + err);
            callback(new Error('Failed to send udp log message. err = ' + err));
        }
    };

    for (var i=0; i<messagesLength; i++) {

        var msg = this.messages[i];
        msg.token = this.token;
        var buff = new Buffer(stringifySafe(msg));

        this._debug('Starting to send messages via udp.');
        this.udpClient.send(buff, 0, buff.length, this.port, this.host, udpSentCallback);
    }
};
```
Look at `udpSentCallback`, it operates with `this` keyword in invalid context. It also references on undefined `callback`.

```javascript
LogzioLogger.prototype.close = function () {
    // clearing the timer allows the node event loop to quit when needed
    clearTimeout(this.timer);
    if (this.protocol === 'udp') {
        this.udpClient.close();
    }

    // send pending messages, if any
    if (this.messages.length > 0) {
        this._debug("Closing, purging messages.");
        this._popMsgsAndSend();
    }

    // no more logging allowed
    this.closed = true;
};
```

`close` method closes UDP socket first, and right after that it tries to send buffered messages. In this case buffered messages won't send and error will thrown by invalid `udpSentCallback`.

**Comment by AlonMiz** [comment]
@sempasha fixed in v0.4.14
thank you
